### PR TITLE
Implement stream insert test case

### PR DESCRIPTION
**Pull request by kevindiu**
Signed-off-by: kevindiu <kevindiujp@gmail.com>

### Description:

This PR implements stream insert test for pkg/agent/ngt/handler.
It also implements server stream mock for testing.

### Environment:

- Go Version: 1.18.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.14.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/test/data/request/object_test.go [hunk 1]
```diff
@@ -151,3 +152,143 @@ func TestGenObjectLocations(t *testing.T) {
...
+			checkFunc := test.checkFunc
+			if test.checkFunc == nil {
+				checkFunc = defaultCheckFunc
+			}
+
+			got := GenObjectStreamLocation(test.args.num, test.args.name, test.args.ipAddr)
+			if err := checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/test/data/request/object_test.go:291]:
```suggestion
```

File: pkg/agent/core/ngt/handler/grpc/insert_test.go [hunk 2]
```diff
@@ -1408,100 +1423,1288 @@ func Test_server_Insert(t *testing.T) {
...
+				want: want{
+					rpcResp: request.GenObjectStreamLocation(1, name, ip),
+				},
+			}
+		}(),
+		func() test {
+			reqs := []*payload.Insert_Request{
+				{
+					Vector: &payload.Object_Vector{
+						Id:     "uuid-1",
```
Comment (vankichi) [inline pkg/agent/core/ngt/handler/grpc/insert_test.go:1860]: You can define `uuid-1` as the default value.
Many test cases use it.

**Review by vankichi — CHANGES_REQUESTED** [review]

File: pkg/agent/core/ngt/handler/grpc/insert_test.go (hunk 2 above)
Comment (kevindiu) [reply pkg/agent/core/ngt/handler/grpc/insert_test.go]: fixed :)

File: pkg/agent/core/ngt/handler/grpc/insert_test.go [hunk 3]
```diff
@@ -1408,100 +1423,1289 @@ func Test_server_Insert(t *testing.T) {
...
+						recvIdx++
+
+						return nil
+					},
+					SendMsgFunc: func(i interface{}) error {
+						rpcResp = append(rpcResp, i.(*payload.Object_StreamLocation))
+						return nil
+					},
+				},
+			}
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply pkg/agent/core/ngt/handler/grpc/insert_test.go:2686]: It mocks the stream server implementation.
The `RecvMsg()` will keep receiving the insert request until it return the error.
It keep receiving the insert request until all insert request is processed (line 2668 - 2670)

Ref: 
https://github.com/vdaas/vald/blob/master/pkg/agent/core/ngt/handler/grpc/insert.go#L139
https://github.com/vdaas/vald/blob/a41df5e0b8ab31de9185f9b51479a271245756d9/internal/net/grpc/stream.go#L97-L105

**Review by hlts2 — APPROVED** [review]

**Pull request merged**